test/functional/mining_mainnet.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test mining on an alternate mainnet
+
+Test mining related RPCs that involve difficulty adjustment, which
+regtest doesn't have. It uses an alternate mainnet chain.
+
+For easier testing the difficulty is maximally increased in the first (and only)
+regarget period, by producing blocks approximately 2 minutes apart.
+
+The alternate mainnet chain was generated as follows:
+- coinbase address derived from first BIP32 test vector master key:
+  tr(xprv9s21ZrQH143K3QTDL4LXw2F7HEK3wJUD2nW2nRk4stbPy6cq3jPPqjiChkVvvNKmPGJxWUtg6LnF5kejMRNNU3TGtRBeJgk33yuGBxrMPHi/86h/0h/0h/0/0)#d6vpenmd
+- use faketime to set node clock to 2 minutes after genesis block
+- mine a block using a CPU miner such as https://github.com/pooler/cpuminer
+- restart node with a faketime 2 minutes later
+
+for i in {1..2015}
+do
+ faketime "`date -d @"$(( 1231006505 + $i * 120 ))"  +'%Y-%m-%d %H:%M:%S'`" \
+ bitcoind -connect=0 -nocheckpoints -stopatheight=$i
+done
+
+The CPU miner is kept running as follows:
+
+./minerd --coinbase-addr ... --no-stratum --algo sha256d --no-longpoll --scantime 3 --retry-pause 1
+
+This makes each block determinisic except for its timestamp and nonce, which
+are stored in data/mainnet_alt.json and used to reconstruct the chain without
+having to redo the proof-of-work.
+
+The timestamp was not kept constant because at difficulty 1 it's not sufficient
+to only grind the nonce. Grinding the extra_nonce or version field instead
+would have required additional (stratum) software. It would also make it more
+complicated to reconstruct the blocks in this test.
```

Comment:
Wonder if it would be sufficient to grind only the nonce if starting with a testnet genesis block (instead of mainnet), but not sure it's worth digging into. It might allow removing the timestamps from `mainnet_alt.json` (reducing 35KB of the current 71KB), but the current approach is already more manageable that the previous ~1MB.

Reply:
Testnet3, testnet4 and mainnet all have a difficulty 1 genesis block, so it wouldn't make a difference.